### PR TITLE
New Source: Add New Arabic Source: rewayat.club

### LINKED ISSUE #78: Rewayat Club
### Source name

Rewayat Club

### Source link

https://rewayat.club/

### Language

Arabic

### Other details

_No response_

### PR DESCRIPTION
**Pull request by Dredd1987**
Added a new source plugin for rewayat.club, providing access to Arabic novels. This PR resolves issue (close #78)

### CODE REVIEW

**Comment by Dredd1987** [comment]
I apologize for the inconvenience earlier.

File: src/plugins/arabic/rewayatclub.ts
Comment (rajarsheechatterjee) [inline src/plugins/arabic/rewayatclub.ts:1]: We could remove the `console.log`.

**Comment by Dredd1987** [comment]
done, and there are some changes to the ao3 plugin that add chapter release dates.

**Comment by nyagami** [comment]
Default this site return chapter list with descending ordering which is not handled properly in app.
I notice that a cookie named auth.novelChaptersOrdering is used to change ordering.
Could you try to add it in request and check if it works
image

**Comment by Dredd1987** [comment]
I tried it, but I couldn't get it to work. It functions properly on the website when the cookie is set, but not so much in the plugin.

**Comment by K1ngfish3r** [comment]
```ts
  parseChapters(data: ChapterData, novelPath: string) {
    const chapter: Plugin.ChapterItem[] = [];
    data.results.map((item: ChapterEntry) => {
      chapter.push({
        name: item.title,
        releaseTime: new Date(item.date).toISOString(),
        path: `${novelPath}/${item.number}`,
        chapterNumber: item.number,
      });
    });
    return chapter;
  }
  async parsePage(novelPath: string, page: string): Promise<Plugin.SourcePage> {
    const pagePath = novelPath.slice(6);
    const pageUrl = `https://api.rewayat.club/api/chapters/${pagePath}/?ordering=number&page=${page}`;
    const dataJson = await fetchApi(pageUrl).then(r => r.json());
    const chapters = this.parseChapters(dataJson, novelPath);
    return {
      chapters,
    };
  }

**Review by nyagami — APPROVED** [review]

**Pull request merged**